benchmarks/hash_table/static_multimap/insert_bench.cu

```diff
@@ -21,37 +21,41 @@
 #include <nvbench/nvbench.cuh>
 
 #include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/transform.h>
 
 /**
  * @brief A benchmark evaluating multi-value `insert` performance:
  * - Total number of insertions: 100'000'000
  * - CG size: 8
  */
-template <typename Key, typename Value, dist_type Dist, nvbench::int32_t Multiplicity>
+template <typename Key, typename Value, typename Dist>
 std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> nvbench_static_multimap_insert(
-  nvbench::state& state,
-  nvbench::type_list<Key, Value, nvbench::enum_type<Dist>, nvbench::enum_type<Multiplicity>>)
+  nvbench::state& state, nvbench::type_list<Key, Value, Dist>)
 {
+  using pair_type = cuco::pair_type<Key, Value>;
+
   auto const num_keys  = state.get_int64("NumInputs");
   auto const occupancy = state.get_float64("Occupancy");
 
   std::size_t const size = num_keys / occupancy;
 
-  std::vector<Key> h_keys(num_keys);
-  std::vector<cuco::pair_type<Key, Value>> h_pairs(num_keys);
-
-  generate_keys<Dist, Multiplicity, Key>(h_keys.begin(), h_keys.end());
+  thrust::device_vector<Key> keys(num_keys);
 
-  for (auto i = 0; i < num_keys; ++i) {
-    Key key           = h_keys[i];
-    Value val         = h_keys[i];
-    h_pairs[i].first  = key;
-    h_pairs[i].second = val;
-  }
+  key_generator gen;
+  gen.generate<Dist>(state, thrust::device, keys.begin(), keys.end());
 
-  thrust::device_vector<cuco::pair_type<Key, Value>> d_pairs(h_pairs);
+  thrust::device_vector<pair_type> pairs(num_keys);
+  thrust::transform(thrust::device,
+                    keys.begin(),
+                    keys.end(),
+                    pairs.begin(),
+                    [] __host__ __device__(Key const& key) {
+                      return thrust::raw_reference_cast(pair_type(key, 42));
+                    });
```

Comment:
This changed the original behavior of matching rate control. Should we use `dropout` here?

Reply:
Can you elaborate some more? How did the behavior change, and where do we need `dropout` in ìnsert_bench`?